src/test/scala/CounterSpecification.scala

```diff
@@ -0,0 +1,36 @@
+import Cqrs.Aggregate._
+import Cqrs.Database.EventSerialisation
+import Domain.Counter
+import org.scalacheck.{Properties, Gen}
+
+object CounterSpecification extends Properties("Counter") {
+
+  property("fails when below zero") = CounterSpec.property()
+
+  private object CounterSpec extends AggregateCommands[Counter.Event, Counter.flow.StateData, Int] {
+
+    override val genInitialState = Gen.const(0)
+    override def initSutActions = Counter.CounterAggregate.initAggregate(AggregateId("test counter"))
+
+    def genCommand(state: State): Gen[Command] =
+      Gen.oneOf(
+        Gen.const(Increment),
+        Gen.const(Decrement)
+      )
+
+    override def evtSerializer = implicitly[EventSerialisation[Counter.Event]]
+
+    case object Increment extends AggregateCommand {
+      def commandActions = Counter.CounterAggregate.handleCommand(Counter.Increment)
+      def nextState(state: State) = state + 1
```

Comment:
Yes... After implementing this commands yest model I think it's not as useful as I initially  thought.. 

Simple quickcheck cases for specific domain should be much clearer.. What do you think?